src/routes/concepts/stores.mdx

```diff
@@ -87,14 +89,75 @@ const App = () => {
 }
 ```
 
+When you create a store, it contains the initial state, but _no signals_ are immediately created to track changes to that state.
+These signals are created lazily, meaning they come into existence only when you access the data within a reactive context.
+
+When you use this data within a reactive context, such as in a template, computed property, or effect, that is when the signals are created and dependencies are established.
+
+For example, if you wanted to you wanted to print out every new user, adding the 
+line below won't work because it is not within a tracked scope.
+
+
+```tsx ins={9}
+const App = () => {
+	const [store, setStore] = createStore({
+		userCount: 3,
+		users: [ ... ],
+	})
+
+	const addUser = () => { ... }
+
+	console.log(store.users.at(-1)) // This won't work
+
+	return (
+		<div>
+			<h1>Hello, {store.users[0].username}</h1>
+			<p>User count: {store.userCount}</p>
+      <button onClick={addUser}>Add user</button>
+		</div>
+	)
+}
+```
+
+Rather, you would have to add this in a tracking scope, such as within a [`createEffect`](/reference/basic-reactivity/create-effect) so that a dependency is established.
+
+```tsx del={9} ins={10-12}
+const App = () => {
+	const [store, setStore] = createStore({
+		userCount: 3,
+		users: [ ... ],
+	})
+
+	const addUser = () => { ... }
+
+	console.log(store.users.at(-1))
+	createEffect(() => {
+		console.log(store.users.at(-1))
+	})
+
+	return (
+		<div>
+			<h1>Hello, {store.users[0].username}</h1>
+			<p>User count: {store.userCount}</p>
+      <button onClick={addUser}>Add user</button>
+		</div>
+	)
+}
+```
+
 ## Modifying store values
 
 Updating values within a store is best accomplished using a setter provided by the `createStore` initialization.
 This setter allows for the modification of a specific key and its associated value, following the format `setStore(key, newValue)`:
 
-```jsx
-setStore("users", (otherUsers) => [
-	...otherUsers,
+```jsx "setStore"
+const [store, setStore] = createStore({
+	userCount: 3,
+	users: [ ... ],
```

Comment:
What is this trying to convey?

Reply:
@LadyBluenotes it's just a collapsed version of the examples above to reduce space taken up by the repetitive content. It is pretty similar to how editors actually show collapsed lines.

![image](https://github.com/solidjs/solid-docs-next/assets/25469167/af78c1fe-f567-4c65-a2f3-3726d3fbaef3)